playbooks/group_vars/libvirt_guests.yaml

```diff
@@ -10,15 +10,26 @@ vm_hostvars: '{{ hostvars[vm_host] }}'
 # (i.e.: all VM definitions), as they create circular dependencies that cannot
 # be resolved.
 
-# Connection values.
-ansible_connection: community.libvirt.libvirt_qemu
-ansible_libvirt_uri: qemu+ssh://{{ vm_hostvars.ansible_host }}/system
-ansible_host: '{{ vm_name }}'
+# Connect using QEMU guest agent.
+libvirt__vm_use_qemu_connection: true
+ansible_connection: |-
+  {{
+    'community.libvirt.libvirt_qemu' if libvirt__vm_use_qemu_connection else
+    'ansible.builtin.ssh'
+  }}
+ansible_libvirt_uri: |-
+  qemu+ssh://{{
+    vm_hostvars.get('ansible_ssh_host') or
+    vm_hostvars.get('ansible_host') or
+    vm_hostvars.get('inventory_hostname')
+  }}/system
+ansible_host: '{{ vm_name if libvirt__vm_use_qemu_connection else host_fqdn }}'
 
 bootstrap__install_packages__libvirt:
   - ifupdown
   # Needed for ansible/libvirt connection.
   - qemu-guest-agent
+  - openssh-server
```

Comment:
(nit: Does the comment above, "Needed for ansible/libvirt connection", applies to qemu-guest-agent only, or openssh-server too? If only qemu-guest-agent, could you move openssh-server two lines up, before the comment?)